src/GhcUtil.hs

```diff
@@ -58,7 +62,18 @@ handleDynamicFlags flags = do
 #else
   let parseDynamicFlags' = parseDynamicFlags
 #endif
-  (dynflags, locSrcs, _) <- (setHaddockMode `fmap` getSessionDynFlags) >>= (`parseDynamicFlags'` flags)
+  (dynflags', locSrcs, _) <- (setHaddockMode `fmap` getSessionDynFlags) >>= (`parseDynamicFlags'` flags)
+
+#if __GLASGOW_HASKELL__ >= 910
+  -- never display error links when displaying exception. Otherwise we get
+  -- failures like this:
+  --    expected: "Foo.hs:6:1: error: [GHC-58481]\n    parse error..."
+  --    but got: "Foo.hs:6:1: error: [\ESC]8;;https://errors.haskell.org/messages/GHC-58481\ESC\\GHC-58481\ESC]8;;\ESC\\]\n    parse error..."
+  let dynflags = dynflags' { useErrorLinks = Never }
```

Comment:
@hsyl20 we need to differentiate between:

1. Extracting comments with the GHCAPI.
2. Evaluating expressions with GHCi.

For (1) we want the exact same error behavior as GHC, including colors and hyperlinks.

For (2) we don't want colors and hyperlinks.54 OK